renn/data/tokenizers.py

```diff
@@ -1,11 +1,67 @@
 """Text processing."""
 
-import tensorflow_datasets as tfds
-import tensorflow_text as tft
+from collections import Counter
+import itertools
 
+from renn.data import wordpiece_tokenizer_learner_lib as learner
 
-build_vocab = tfds.features.text.SubwordTextEncoder.build_from_corpus
-load_vocab = tfds.features.text.SubwordTextEncoder.load_from_file
+import tensorflow_text as text
+import tensorflow as tf
 
 
+# Special tokens
+_JOINER = '##'
+_UNK = '<unk>'
+_CLS = '<cls>'
+_SEP = '<sep>'
 
+
+def build_vocab(corpus_generator, split_fun=str.split, vocab_size=2 ** 15):
+  """Builds a vocab file from a text generator."""
+
+  # Split documents into words.
+  words = itertools.chain(*map(split_fun, corpus_generator))
```

Comment:
This works for small/medium size training data, but could face performance issue if we need to scale up. But not something to worry about now.

Reply:
agreed, I'll add a note about this in the file.